src/extension-runners/firefox-desktop.js

```diff
@@ -0,0 +1,229 @@
+/* @flow */
+
+// Import flow types from npm dependencies.
+import type FirefoxProfile from 'firefox-profile';
+
+import {
+  MultipleExtensionsReloadError,
+  RemoteTempInstallNotSupported,
+  WebExtError,
+} from '../errors';
+import * as defaultFirefoxApp from '../firefox';
+import {
+  connectWithMaxRetries as defaultFirefoxConnector,
+} from '../firefox/remote';
+import {createLogger} from '../util/logger';
+// Import flow types from project files.
+import type {
+  FirefoxRDPResponseAddon,
+  RemoteFirefox,
+} from '../firefox/remote';
+import type {
+  ExtensionRunnerParams,
+} from './base';
+import type {
+  FirefoxPreferences,
+} from '../firefox/preferences';
+import type {
+  FirefoxInfo, // eslint-disable-line import/named
+} from '../firefox/index';
+
+export type FirefoxDesktopExtensionRunnerParams = ExtensionRunnerParams & {
```

Comment:
actually, it looks that we can do something that works similar to this and which also supports the exact types:

- https://github.com/facebook/flow/issues/2626#issuecomment-290107574

I'm going to try this approach as part of the next round of updates on this PR.